driver-core/src/main/com/mongodb/internal/connection/IdHoldingBsonWriter.java

```diff
@@ -55,6 +55,7 @@
     private BasicOutputBuffer outputBuffer;
     private String currentFieldName;
     private BsonValue id;
+    private boolean idFieldIsNested = false;
```

Comment:
Not sure what "nested" means in this context.  Can you clarify?  When I first read it, I was thinking it meant that there was an _id field nested inside a subdocument, but that clearly is not right. Maybe we can come up with a better name.

Reply:
Its tricky - it really means the id field is a top level array and has to be nested in a document to decode.  Happy for any suggestions.